### DISCUSSION TITLE
shouldideploy on Slack!

### DISCUSSION THREAD

**Discussion by Brunomachadob** [baires/shouldideploy#467] [Ideas]
Hey there! First of all, thanks a lot for the great work, this website is awesome, and to improve it even more I was wondering, what do you think about having a slack app for it? It would be great to simply type `/shouldideploy` on slack and get a response immediately on the channel.

I've just did a really simple version of it just to try it locally and it worked pretty fine. We could probably include this idea on your existing server. You can check it out here: https://github.com/Brunomachadob/shouldideploy-slack

It should be a matter of only returning the api response in the format slack expects, you can see it here: https://github.com/Brunomachadob/shouldideploy-slack/blob/88751dfe63093bac0e661dc63e570f9473d2efe8/index.js#L44

A `Yes` example:
![image](https://user-images.githubusercontent.com/4688763/115783619-5f6c1f80-a3bd-11eb-8182-5a36d47dcd65.png)

A `No` example:
![image](https://user-images.githubusercontent.com/4688763/115783864-aeb25000-a3bd-11eb-9d74-60e2dfc1d3a3.png)

An error example (we probably don't need this if we embed it here on the same server 🤷🏻‍♂️ ):
![image](https://user-images.githubusercontent.com/4688763/115783717-7c085780-a3bd-11eb-9cb8-fc7ed28121b5.png)

Let me know what you think!

Thanks!

**Comment by baires** [comment]
This is great 👏  👏  👏 

May you can create a new endpoint under [api](https://github.com/baires/shouldideploy/tree/master/pages/api) to return a formatted payload for Slack. PR always welcome

**Reply by Brunomachadob** [reply]
Sure, glad you like it. I'll prepare and push it.

**Reply by Brunomachadob** [reply]
PR opened: https://github.com/baires/shouldideploy/pull/486

**Reply by Brunomachadob** [reply]
Hey @baires, was the change successfully deployed? If I query https://shouldideploy.today/api/slack?tz=UTC, it rewrites to the old API. Locally this does not happen, maybe there is something else that we need to change?

**Reply by baires** [reply]
I noticed that today. Its now fixed on https://github.com/baires/shouldideploy/pull/488

**Reply by Brunomachadob** [reply]
Interesting, I was expecting the matching to happen from top to bottom on this list and return on the first match 🤔 
I'll need to read more about Vercel later to understand this better.

Or maybe could this be related to browser cache also?
Even after your PR, it only worked for me after reseting the cache and hard reloading.

**Reply by baires** [reply]
Looks like a cache issue around vercel. I just curl https://shouldideploy.today/api/slack and works. Also https://shouldideploy.today/api/slack?_vercel_no_cache=1 does the trick. Will read more about this tomorrow
